etc/scripts/analysis/user-activity.js

```diff
@@ -0,0 +1,510 @@
+ */
+
+/**
+ * An analysis script designed to determine how active user accounts are for a tenant.
+ */
+
+var _ = require('underscore');
+var bunyan = require('bunyan');
+var csv = require('csv');
+var fs = require('fs');
+var optimist = require('optimist');
+var path = require('path');
+var util = require('util');
+
+var ActivityDAO = require('oae-activity/lib/internal/dao');
+var AuthzUtil = require('oae-authz/lib/util');
+var Cassandra = require('oae-util/lib/cassandra');
+var log = require('oae-logger').logger('oae-script-main');
+var OAE = require('oae-util/lib/oae');
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var Validator = require('oae-util/lib/validator').Validator;
+
+var argv = optimist.usage('$0 [-t cam] [--config <path/to/config.js>]')
+    .alias('t', 'tenant')
+    .describe('t', 'Specify the tenant alias of the tenant whose users activity to analyze. By default, analyze all tenants')
+
+    .alias('c', 'config')
+    .describe('c', 'Specify an alternate config file')
+    .default('c', 'config.js')
+
+    .alias('h', 'help')
+    .describe('h', 'Show usage information')
+    .argv;
+
+if (argv.help) {
+    optimist.showHelp();
+    return process.exit(1);
+}
+
+// Get the config
+var configPath = path.resolve(process.cwd(), argv.config);
+var config = require(configPath).config;
+var tenantAlias = argv.tenant;
+
+// Ensure that this application server does NOT start processing any preview images
+config.previews.enabled = false;
+
+// Ensure that we're logging to standard out/err
+config.log = {
+    'streams': [
+        {
+            'level': 'info',
+            'stream': process.stdout
+        }
+    ],
+    'serializers': {
+        'err': bunyan.stdSerializers.err,
+        'req': bunyan.stdSerializers.req,
+        'res': bunyan.stdSerializers.res
+    }
+};
+
+var start = Date.now();
+var streamInfo = _createCsvStream();
+var csvStream = streamInfo.csv;
+var fileStream = streamInfo.file;
+var filePath = streamInfo.filePath;
+
+// Spin up the application container. This will allow us to re-use existing APIs
+OAE.init(config, function(err) {
+    if (err) {
+        log().error({'err': err}, 'Unable to spin up the application server');
+        return process.exit(err.code);
+    }
+
+    _groupUsersByEmail(tenantAlias, function(usersByEmail) {
+        var batches = _.chain(usersByEmail)
+            // Only keep profiles that have duplicate emails
+            .filter(function(userHashes) { return (userHashes.length > 1); })
+            .flatten()
+            .tap(function(userHashes) {
+                log().info('Preparing to write a total of %s users who have duplicate emails', userHashes.length);
+            })
+
+            // Re-group things into sane concurrent batches
+            .groupBy(function(userHash, i) { return Math.floor(i / 5); })
+            .values()
+            .value();
+
+        log().info('Begin writing %s batches of user activity to CSV', batches.length);
```

Comment:
Oops, misread `/` as `%` and was getting worries that we'd end up with batches of ~5000 users for a minute. Ignore the above!